host-interaction/registry/open/open-registry-key.yml

```diff
@@ -3,14 +3,29 @@ rule:
     name: open registry key
     namespace: host-interaction/registry/open
     author: michael.hunhoff@fireeye.com
-    scope: function
+    scope: basic block
     mbc:
       - Operating System::Registry::Open Registry Key [C0036.003]
     examples:
       - Practical Malware Analysis Lab 03-02.dll_:0x10004706
       - 493167E85E45363D09495D0841C30648:0x404D60
+      - B5F85C26D7AA5A1FB4AF5821B6B5AB9B:0x4045F2
   features:
     - or:
       - api: advapi32.RegOpenKey
       - api: advapi32.RegOpenKeyEx
+      - api: advapi32.RegCreateKey
```

Comment:
overlap with `create registry key`?

Reply:
`RegCreateKey` can, per MSDN:

> Creates the specified registry key. If the key already exists in the registry, the function opens it.

Because of its dual purpose I included a check for `RegCreateKey` in both `open registry key` and `create registry key`. I had a similar struggle placing dual purpose APIs like this when creating the `create file` and `open file` rules.

Maybe there needs to be a bigger discussion on how dual purpose APIs should be expressed in capa rules e.g. including `RegCreateKey` in both `create registry key` and `open registry key` versus creating a separate rule named something like `create or open registry key` that only includes the dual purpose APIs like `RegCreateKey`. The latter IMO is more concise.

Reply:
Good points. All our current rules use `create` and `open` combined (or should), so combining into a `create or open registry key` rule makes sense in this case.